tests/e2e/vmservice_late_binding.go

```diff
@@ -0,0 +1,291 @@
+	   volume.kubernetes.io/selected-node is set to zone - This should have the zone name where the VM gets provisioned
+	6. Verify CNS metadata for PVC
+	7. Verify PVC's attached to VM
+	8. Validate TotalQuota, StoragePolicyQuota, storageQuotaUsage of VmserviceVm's and PVc's
+
+	*/
+	ginkgo.It("vmserviceVM-WFFC-BasicCase", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var totalQuotaUsedBefore, pvc_storagePolicyQuotaBefore, pvc_storagePolicyUsageBefore *resource.Quantity
+		var vm_totalQuotaUsedBefore, vm_storagePolicyQuotaBefore, vm_storagePolicyUsageBefore *resource.Quantity
+
+		ginkgo.By("Set Storage Quota on namespace")
+		restConfig = getRestConfigClient()
+		framework.Logf("NameSpace : %s", namespace)
+		if isQuotaValidationSupported {
+			ginkgo.By("create resource quota")
+			setStoragePolicyQuota(ctx, restConfig, storagePolicyName, namespace, rqLimit)
+		}
+
+		wffcStorageclass, err := client.StorageV1().StorageClasses().Get(ctx, wffsStoragePolicyName, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("wffsStoragePolicyName: %s wffcStorageclass: %s", wffsStoragePolicyName, wffcStorageclass)
+
+		storageclass, err := client.StorageV1().StorageClasses().Get(ctx, storagePolicyName, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		framework.Logf("wffsStoragePolicyName: %s wffcStorageclass: %s", wffsStoragePolicyName, wffcStorageclass)
+
+		ginkgo.By("Read QuotaDetails Before creating workload")
+		if isQuotaValidationSupported {
+			totalQuotaUsedBefore, _, pvc_storagePolicyQuotaBefore, _, pvc_storagePolicyUsageBefore, _ =
+				getStoragePolicyUsedAndReservedQuotaDetails(ctx, restConfig,
+					storageclass.Name, namespace, pvcUsage, volExtensionName, true)
+
+			vm_totalQuotaUsedBefore, _, vm_storagePolicyQuotaBefore, _, vm_storagePolicyUsageBefore, _ =
+				getStoragePolicyUsedAndReservedQuotaDetails(ctx, restConfig,
+					storageclass.Name, namespace, vmUsage, vmServiceExtensionName, true)
+		}
+
+		ginkgo.By("Create a PVC")
+		pvc, err := createPVC(ctx, client, namespace, nil, "", wffcStorageclass, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			ginkgo.By("Delete PVCs")
+			err = fpv.DeletePersistentVolumeClaim(ctx, client, pvc.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Wait for VM images to get listeed undernamespace and create VM")
+		err = pollWaitForVMImageToSync(ctx, namespace, vmImageName, poll, supervisorClusterOperationsTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		ginkgo.By("Creating VM bootstrap data")
+		secretName := createBootstrapSecretForVmsvcVms(ctx, client, namespace)
+		defer func() {
+			ginkgo.By("Deleting VM bootstrap data")
+			err := client.CoreV1().Secrets(namespace).Delete(ctx, secretName, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating VMs")
```

Comment:
Use method createVmServiceVm()

Reply:
There is update Going on, for createVmServiceVm method. Once that is done I will take those change.